File: docs/WidgetMill/packages/spec/src/nodes/aspectRatio.ts
import { z } from "zod";
import { NodeRef } from "../node-ref";

/** AspectRatio — força uma razão largura/altura no filho (catálogo Flutter). */
export const AspectRatioNode = z.object({
  type: z.literal("aspectRatio"),
  // `aspectRatio` é obrigatório (sem default) — o Inspector marca com `*`.
  props: z.object({
    aspectRatio: z.number(),
  }),
  child: NodeRef.optional(),
});

File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/aspect_ratio.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// AspectRatio — força uma razão largura/altura no filho.
Widget buildAspectRatio(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return AspectRatio(
    aspectRatio: parseDouble(p['aspectRatio']) ?? 1.0,
    child: r.maybeRender(context, node.child),
  );
}
